lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp

```diff
@@ -2312,6 +2312,19 @@ void LifetimeChecker::handleSelfInitUse(unsigned UseID) {
   }
 }
 
+// In case of `var` initializations, SILGen creates a dynamic begin/end_access
+// pair around the initialization store. If it's an initialization (and not
+// a re-assign) it's guanranteed that it's an exlusive access and we can
+// convert the access to an `[init] [static]` access.
+static void setStaticInitAccess(SILValue memoryAddress) {
+  if (auto *ba = dyn_cast<BeginAccessInst>(memoryAddress)) {
+    if (ba->getEnforcement() == SILAccessEnforcement::Dynamic) {
+      ba->setEnforcement(SILAccessEnforcement::Static);
+      if (ba->getAccessKind() == SILAccessKind::Modify)
```

Comment:
@eeckstein I think you might be able to take this even a step further and change SILGen to always emit these as inits, unknown. It would then be an error if access control Can not convert it to static. I did something like this for consuming parameters which use deinit. Anyways, just a friendly thought.

Reply:
I thought about this, too. But SILGen doesn't know yet if it's an init or re-assign. And e.g. in a class initializer you can let self escape (after all members are initialized) and then re-assign a member. I'm not sure if it's even safe to guard such a reassign-after-self-escape as a statically checked access.

Reply:
s/guanranteed/guaranteed